crates/storage/store/engines/in_memory.rs

```diff
@@ -378,6 +378,11 @@ impl StoreEngine for Store {
         Ok(self.inner().canonical_hashes.get(&block_number).cloned())
     }
 
+    fn unset_canonical_block(&self, number: BlockNumber) -> Result<(), StoreError> {
+        self.inner().canonical_hashes.remove(&number);
```

Comment:
Apart from removing this one, it should remove all the ones above?

Reply:
For now I am not doing this as it would require to re-request for the latest block, which we already did, and we only use this in fork choice. I did add a note though, so we can reevaluate if this function starts being used in multiple places.